MongoDbGenericRepository/BaseMongoRepository.Create.cs

```diff
@@ -15,7 +15,7 @@ public interface IBaseMongoRepository_Create : IBaseMongoRepository_Create<Guid>
         /// <typeparam name="TDocument">The type representing a Document.</typeparam>
         /// <typeparam name="TKey">The type of the primary key for a Document.</typeparam>
         /// <param name="document">The document you want to add.</param>
-        Task AddOneAsync<TDocument, TKey>(TDocument document)
+        Task AddOneAsync<TDocument, TKey>(TDocument document, string partitionKey = null)
```

Comment:
I do not see much justification for this, if TDocument is a partioned document is should be handle automatically without needing the `string partitionKey = null` param.

Reply:
I agree to use IPartitionedDocument